### ISSUE TITLE
[BUG] Flashcards don't wait any time before being required to review.

### ISSUE THREAD

**Issue by Malachi-H** [st3v3nmw/obsidian-spaced-repetition#749]
**Describe the bug**
For a few flashcards, no matter how many times I review them, they still show up as flashcards due for review. The metadata for the effected cards does not change.

**To Reproduce**

No idea! Seems to happen to random cards and persist for a random amount of time. 

**Expected behavior**
The flashcard's metadata is meant to update and the flashcard is not supposed to be shown until at least the next day.

**Screenshots**
If applicable, add screenshots to help explain the bug. This could be the logs in the developer console (`CTRL` + `SHIFT` + `i`).

**Versions (please complete the following information):**

-   OS: Microsoft Windows 11 Home
-   Obsidian version: 1.4.14
-   Plugin version: v1.10.1
-   If on desktop, Installer version: '5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) obsidian/1.4.14 Chrome/114.0.5735.289 Electron/25.8.1 Safari/537.36'

**Additional context**
Does not fix the non-updating flashcards:
- Closing the review window and re-opening it
- Completing the flashcards multiple times
- Clicking "Reset Card's Progress"

Sometimes fixes some of the non-updating flashcards (not all of them):
Note: If it doesn't work on a flashcard the first time then it doesn't work at all.
- Restarting Obsidian
- Opening the file containing the flashcard 

Consistently Fixes the non-updating flashcards:
- Manually deleting the html comment (the metadata) for each and every card effected by the issue.

**Comment by ronzulu** [comment]
Hi there

I've just started contributing to this project, and felt that it would benefit from a restructure of the code. 

(If you are a developer, you might be interested in the details at https://github.com/st3v3nmw/obsidian-spaced-repetition/discussions/736)

I have just completed this and am hoping that the project owner will accept the (fairly significant) changes I've made.

The code involved in the issue you raised was very intricate and complex. I've simplified this code (IMO), and wouldn't be surprised if the issue has been resolved.

Lets stay in touch and see what the project owner intends to do.

Cheers
Ronny
